worker/peergrouper/machinetracker.go

```diff
@@ -101,9 +101,13 @@ func (m *machineTracker) SelectMongoHostPort(mongoSpace network.SpaceName) strin
 	defer m.mu.Unlock()
 
 	if mongoSpace != "" {
-		return mongo.SelectPeerHostPortBySpace(m.mongoHostPorts, mongoSpace)
+		addr := mongo.SelectPeerHostPortBySpace(m.mongoHostPorts, mongoSpace)
+		logger.Debugf("machine %q selected address %q by space %q from %v", m.id, addr, mongoSpace, m.mongoHostPorts)
```

Comment:
move all of the logging to this portion, and give it much nicer context so we know
a) what machine is being filtered
b) what mechanism we are filtering by
c) move the address list to the end, since it is arbitrarily long and not as interesting as the final address
d) merge what was multiple lines into a single line with all the relevant info